src/test/java/de/tum/in/www1/artemis/exercise/AthenaResourceIntegrationTest.java

```diff
@@ -61,31 +87,133 @@ protected void initTestCase() {
+        programmingSubmission.setParticipation(participation);
+        programmingSubmissionRepository.save(programmingSubmission);
+        // Create example result
+        var result = new Result();
+        result.setScore(1.0);
+        result.setSubmission(programmingSubmission);
+        result.setParticipation(participation);
+        result.setAssessmentType(AssessmentType.MANUAL);
+        // Create example feedback so that Athena can process it
+        var feedback = new Feedback();
+        feedback.setCredits(1.0);
+        feedback.setResult(result);
+        result.setFeedbacks(List.of(feedback));
+
+        result = resultRepository.save(result);
+        feedbackRepository.save(feedback);
+
+        // Enable Athena for the exercise
+        programmingExercise.setFeedbackSuggestionsEnabled(true);
+        programmingExerciseRepository.save(programmingExercise);
+
+        Result response = request.putWithResponseBody("/api/participations/" + participation.getId() + "/manual-results?submit=true", result, Result.class, HttpStatus.OK);
+
+        // Check that nothing went wrong, even with Athena enabled
+        assertThat(response).as("response is not null").isNotNull();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "template", "solution", "tests" })
+    void testRepositoryExportEndpoint(String repoType) throws Exception {
+        // Enable Athena for the exercise
+        programmingExercise.setFeedbackSuggestionsEnabled(true);
+        programmingExerciseRepository.save(programmingExercise);
+
+        // Add Git repo for export
+        programmingExerciseUtilService.createGitRepository();
+
+        // Get exports from endpoint
+        var authHeaders = new HttpHeaders();
+        authHeaders.add("Authorization", athenaSecret);
+        var repoZip = request.getFile("/api/public/athena/programming-exercises/" + programmingExercise.getId() + "/repository/" + repoType, HttpStatus.OK,
+                new LinkedMultiValueMap<>(), authHeaders, null);
+
+        // Check that ZIP contains file
+        try (var zipFile = new ZipFile(repoZip)) {
+            assertThat(zipFile.size()).as("zip file contains files").isGreaterThan(0);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "template", "solution", "tests" })
+    void testRepositoryExportEndpointsFailWhenAthenaNotEnabled(String repoType) throws Exception {
+        var authHeaders = new HttpHeaders();
+        authHeaders.add("Authorization", athenaSecret);
+
+        // Expect status 503 because Athena is not enabled for the exercise
+        request.get("/api/public/athena/programming-exercises/" + programmingExercise.getId() + "/repository/" + repoType, HttpStatus.SERVICE_UNAVAILABLE, Result.class,
+                authHeaders);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "repository/template", "repository/solution", "repository/tests", "submissions/100/repository" })
```

Comment:
For `testRepositoryExportEndpointsFailWhenAthenaNotEnabled`, no reason. I added it. For `testRepositoryExportEndpoint`, because the setup to test is quite complex and I didn't want to duplicate it. We already test pretty much everything in `AthenaRepositoryExportServiceTest.shouldExportRepository` for this, so I'll skip this.